src/services/textChanges.ts

```diff
@@ -365,6 +388,36 @@ namespace ts.textChanges {
             this.insertNodeAt(sourceFile, endNode.end, type, { prefix: ": " });
         }
 
+        public tryInsertJSDocType(sourceFile: SourceFile, node: Node, type: TypeNode): void {
+            const printed = createPrinter().printNode(EmitHint.Unspecified, type, sourceFile);
+            let commentText;
+            if (isGetAccessorDeclaration(node)) {
+                commentText = ` @return {${printed}} `;
+            }
+            else {
+                commentText = ` @type {${printed}} `;
+                node = node.parent;
+            }
+            this.insertCommentThenNewline(sourceFile, getLineAndCharacterOfPosition(sourceFile, node.getStart()).character, node.getStart(), commentText);

```

Comment:
`node.getStart(sourceFile)`

Reply:
What does sourceFile do here?

Reply:
`node.getStart()` needs the source file for its text -- it then walks the text starting at `node.pos` to the first non-comment/non-whitespace character. If the SourceFile isn't passed it has to climb ancestors to get it.